ANMN/LTSP/TSaggregator/test/geoserverCatalog.py

```diff
@@ -0,0 +1,98 @@
+                           [-out OUTFILELIST]
+
+optional arguments:
+  -h, --help          show this help message and exit
+  -var VAR            name of the variable of interest, like TEMP
+  -site SITE          site code, like NRMMAI
+  -ft FEATURETYPE     feature type, default timeseries
+  -fv FV              file version, 0, 1 or 2 only
+  -realtime REALTIME  real time data wanted? True or False. Default False
+  -ts TIMESTART       start time like 2015-12-01. Default 1944-10-15
+  -te TIMEEND         end time like 2018-06-30. Default today's date
+  -out OUTFILELIST    name of the file to store the selected files urls.
+                      Default: fileList.csv
+"""
+
+
+import sys
+import argparse
+from datetime import datetime, timedelta
+from distutils.util import strtobool
+
+from owslib.wfs import WebFeatureService
+from owslib.fes import *
+from owslib.etree import etree
+import pandas as pd
+
+
+
+def args():
+    parser = argparse.ArgumentParser(description="Get a list of urls from the AODN geoserver")
+    parser.add_argument('-var', dest='varname', help='name of the variable of interest, like TEMP', required=True)
+    parser.add_argument('-site', dest='site', help='site code, like NRMMAI',  required=True)
+    parser.add_argument('-ft', dest='featuretype', help='feature type, default timeseries', default='timeseries', required=False)
+    parser.add_argument('-fv', dest='fileversion', help='file version 1 or 2 only', default=1, type=int, required=False)
+    parser.add_argument('-ts', dest='timestart', help='start time like 2015-12-01. Default 1944-10-15', default='1944-10-15', required=False)
+    parser.add_argument('-te', dest='timeend', help='end time like 2018-06-30. Default today\'s date', default=str(datetime.now())[:10], required=False)
+    parser.add_argument('-out', dest='outFileList', help='name of the file to store the selected files urls. Default: filelist.csv', default="", required=False)
+    parser.add_argument('-realtime', dest='realtime', help='real time data wanted?', required=False)
+
+    vargs = parser.parse_args()
+    return(vargs)
+
+
+def get_urls(varname, site, featuretype='timeseries', fileversion=1, realtime='False', timestart='1944-10-15', timeend=str(datetime.now())[:10], outFileList=''):
+
+    """
+    get the urls from the geoserver moorings_all_map collection
+    based on user defined filters
+    """
+    serverurl = 'http://geoserver-123.aodn.org.au/geoserver/wfs'
+    ftype = 'imos:moorings_all_map'
+    wfs = WebFeatureService(serverurl, version='1.1.0')
+
+    ## filter first by site
+    filter_site = PropertyIsLike(propertyname='site_code', literal=site)
+    filter_xml = etree.tostring(filter_site.toXML()).decode("utf-8")
+    response = wfs.getfeature(typename=ftype, filter=filter_xml,  outputFormat='csv')
+
+    df = pd.read_csv(response)
+
+    ## then filter by the rest of criteria
+    criteria_var = df.variables.str.contains(varname, regex=False)
```

Comment:
Have to be careful with sub-string matching here. E.g. if varname is "TEMP", you should be looking for sub-string " TEMP," otherwise you might also get variables with names like "XTEMP" or "TEMP_SOMETHING_ELSE".

Reply:
Yes this is a problem indeed. Not only for the extra chars after or before the variable name but with the position of the name in the `variables` string (', TEMP,' the most frequent, 'TEMP, 'if at the beginning, ', TEMP' if at the end. So comma as part of the search string does not solve the problem.

I found a solution using a regex with a word boundary in the Series.str.contains method:

`str.contains('.*\\bTEMP\\b.*', regex=True) `